src/numa_virtual.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * =======================================================================================
+ *
+ *      Filename:  numa_virtual.c
+ *
+ *      Description:  Virtual/Fake NUMA backend
+ *
+ *      Version:   <VERSION>
+ *      Released:  <DATE>
+ *
+ *      Author:   Thomas Gruber (tr), thomas.roehl@googlemail.com
+ *                Tobias Auerochs, tobi291019@gmail.com
+ *      Project:  likwid
+ *
+ *      Copyright (C) 2016 RRZE, University Erlangen-Nuremberg
+ *
+ *      This program is free software: you can redistribute it and/or modify it under
+ *      the terms of the GNU General Public License as published by the Free Software
+ *      Foundation, either version 3 of the License, or (at your option) any later
+ *      version.
+ *
+ *      This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *      WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *      PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License along with
+ *      this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * =======================================================================================
+ */
+
+/* #####   HEADER FILE INCLUDES   ######################################### */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <error.h>
+
+#include <numa.h>
+#include <topology.h>
+
+/* #####   FUNCTION DEFINITIONS  -  LOCAL TO THIS SOURCE FILE   ########### */
+
+static uint64_t
+getFreeNodeMem(void)
+{
+    FILE *fp;
+    uint64_t free = 0;
+    bstring freeString  = bformat("MemFree:");
+    int i;
+
+    if (!access("/proc/meminfo", R_OK))
+    {
+        if (NULL != (fp = fopen ("/proc/meminfo", "r")))
+        {
+            bstring src = bread ((bNread) fread, fp);
+            struct bstrList* tokens = bsplit(src,(char) '\n');
+            for (i=0;i<tokens->qty;i++)
+            {
+                if (binstr(tokens->entry[i],0,freeString) != BSTR_ERR)
+                {
+                     bstring tmp = bmidstr (tokens->entry[i], 10, blength(tokens->entry[i])-10  );
+                     bltrimws(tmp);
+                     struct bstrList* subtokens = bsplit(tmp,(char) ' ');
+                     free = str2int(bdata(subtokens->entry[0]));
+                     bdestroy(tmp);
+                     bstrListDestroy(subtokens);
+                }
+            }
+            bstrListDestroy(tokens);
+            bdestroy(src);
+            fclose(fp);
+        }
+    }
+    else
+    {
+        bdestroy(freeString);
+        ERROR;
+    }
+    bdestroy(freeString);
+    return free;
+}
+
+static uint64_t
+getTotalNodeMem(void)
```

Comment:
Put the function in `src/numa.c`, export it via `src/includes/numa.h` and use it here and in the other `numa_*` files.